lib/core/webdriver/firefox.js

```diff
@@ -211,6 +211,14 @@ module.exports.configureBuilder = function(builder, options) {
     profile.setPreference(pref, defaultFirefoxPreferences[pref]);
   });
 
+  if (options.firefox && options.firefox.preference) {
```

Comment:
adding just before so we can override all values but still all prefreences on the same place.